RELEASE-howto.md

```diff
@@ -0,0 +1,121 @@
+# Howto release PyWPS
+
+This document gives you, as PyWPS release master complete tutorial of how to get
+PyWPS release rolled up and deployed to target server, create packages etc.
+
+## PyWPS versioning
+
+PyWPS uses [Debian version naming system](https://www.debian.org/doc/debian-policy/ch-controlfields.html#s-f-Version). Every policy should be checked against it.
+
+PyWPS uses 3 numbers release description: MAJOR.MINOR.MAINTENANCE. Currently,
+the MAJOR version is `4`. Within MAJOR releases, we should aim, not to break
+backwards compatibility.
+
+Odd MINOR version numbers (0, 2, 4, 6, ...) are considered as stable, where as
+even numbers (1, 3, 5, 7, ...) are current development branches. MINOR releases
+should add new features.
+
+MAINTENANCE number should be used for bugfix releases only. No new features are
+added.
+
+For release candindates `MAJOR.MINOR.MAINTENANCE-rcX` format should be used.
+
+## Create new ticket in repository
+
+At https://github.com/geopython/pywps/issues/new create new ticket with the
+title
+
+```
+PyWPS MAJOR.MINOR.MAJOR-rcX release
+```
+
+At the moment, PyWPS uses `4` as MAJOR version number.
+
+Create new branch in your private repository
+
+```
+git checkout master # or checkout to other point in history, where you need to create new release
+git checkout -b TICKET_NUMBER_pywps-X.Y.Z-rcX-release
```

Comment:
@ldesousa as already noted: Having branch for MAJOR.MINOR versions should enable us to apply (merge) bug fix to current stable releases - the development tree can move one and we can still be able to support older (still supported) versions  - or "current stable".